### PR TITLE
Always creates `SymbolToField "id"` instance

### PR DESCRIPTION
**Pull request by hw202207**
## Context

- Regardless the type of primary key, there is always a `{EntityName}Id` type hence I think it's safe to always create a `SymbolToField` instance for ID.

### CODE REVIEW

File: persistent/ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog for persistent
 
+## Unreleased
+
+* [1497](https://github.com/yesodweb/persistent/pull/1497)
+    * Always generates `SymbolToField "id"` instance
```
Comment (parsonsmatt) [inline persistent/ChangeLog.md:6]: This will need to be a minor version bump - may be best to roll up with #1503

Comment (hw202207) [reply persistent/ChangeLog.md:6]: Sounds reasonable to me. Anything I need to do / pay attention to?

Comment (hw202207) [reply persistent/ChangeLog.md:6]: I'll rebase after #1503 has been merged (adjust the changelog section)

**Comment by parsonsmatt** [comment]
This builds and tests just fine locally so I am going to merge. Thank you!

**Pull request merged**